dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -98,6 +98,40 @@ object -> {
     Some((req.method, req.pathInfo))
 }
 
+object ->> {
+  private val allMethods = Method.all.toSet
+
+  /** Extractor to match an http resource and then enumerate all supported methods:
+    * {{{
+    *   (request.method, Path(request.path)) match {
+    *     case withMethod ->> Root / "test.json" => withMethod {
+    *       case Method.GET => ...
+    *       case Method.POST => ...
+    * }}}
+    *
+    * Returns an error response if the method is not matched, in accordance with [[https://datatracker.ietf.org/doc/html/rfc7231#section-4.1 RFC7231]]
+    */
```

Comment:
Would it be better to put this doc on the object?  I'm not sure what makes it more visible in tooling.

Reply:
Not sure. I copied the pattern from the other matchers, but I can look into it. Generally I don't think extractors play great with tooling.